schema/catvrs/catvrs-source.yaml

```diff
@@ -13,296 +13,254 @@ namespaces:
-  # categorical variation.
 
   # =============================================================================
-  # Top-level Categorical Variation concepts
+  # Constraint Model
   # =============================================================================
 
-  CategoricalVariation:
-    inherits: gks.common:DomainEntity
+  Constraint:
+    maturity: draft
+    type: object  # Maybe we don't want this for abstract class?
+    inherits: vrs:ValueObject  # ditto above
     description: >-
-      A representation of a categorically-defined domain for variation, in which individual 
-      contextual variation instances may be members of the domain.
-    oneOf:
-      - $ref: "#/$defs/CanonicalAllele"
-      - $ref: "#/$defs/CategoricalCnv"
-      - $ref: "#/$defs/DescribedVariation"
-      - $ref: "#/$defs/ProteinSequenceConsequence"
+      Constraints are used to construct an intensional semantics of categorical variant types.
     heritableProperties:
-      members:
-        type: array
-        ordered: false
+      description:
+        extends: description
         description: >-
-          A non-exhaustive list of VRS variation contexts that satisfy the constraints of
-          this categorical variant.
-        items:
-          oneOf:
-            - $refCurie: vrs:Variation
-            - $refCurie: gks.common:IRI
-
-  ProteinSequenceConsequence:
-    maturity: draft
-    type: object
-    inherits: CategoricalVariation
-    description: >-
-      A change that occurs in a protein sequence as a result of genomic changes. Due to the degenerate nature
-      of the genetic code, there are often several genomic changes that can cause a protein sequence consequence.
-      The protein sequence consequence, like a :ref:`CanonicalAllele`, is defined by an
-      `Allele <https://vrs.ga4gh.org/en/2.0/terms_and_model.html#variation>` that is representative of a collection
-      of congruent Protein Alleles that share the same altered codon(s).
-    properties:
-      type:
-        extends: type
-        const: ProteinSequenceConsequence
-        default: ProteinSequenceConsequence
-        description: 'MUST be "ProteinSequenceConsequence"'
-      definingContext:
-        oneOf:
-          - $refCurie: vrs:Allele
-          - $refCurie: gks.common:IRI
+          A textual description of the domain of variation that should match the categorical 
+          variation entity.
+      label:
+        extends: label
         description: >-
-          The `VRS Allele <https://vrs.ga4gh.org/en/2.0/terms_and_model.html#allele>`_
-          object that is congruent with (projects to the same codons) as alleles on other protein reference 
-          sequences.
+            A primary label for the categorical variation. This required property should provide a 
+            short and descriptive textual representation of the concept.
     required:
-      - definingContext
+      - label
```

Comment:
This is interesting. Do we expect that all implementations will need to label all constraints?

What is an example of this from, e.g. CIViC?

Reply:
I concede, labels that probably shouldn't be required.